contracts/0.4.24/Lido.sol

```diff
@@ -111,151 +81,162 @@ contract Lido is ILido, StETH, AragonApp {
     /// Not used in the logic
     bytes32 internal constant TOTAL_EL_REWARDS_COLLECTED_POSITION = keccak256("lido.Lido.totalELRewardsCollected");
 
-    /// @dev Credentials which allows the DAO to withdraw Ether on the 2.0 side
-    bytes32 internal constant WITHDRAWAL_CREDENTIALS_POSITION = keccak256("lido.Lido.withdrawalCredentials");
-
-    /**
-    * @dev As AragonApp, Lido contract must be initialized with following variables:
-    * @param _depositContract official ETH2 Deposit contract
-    * @param _oracle oracle contract
-    * @param _operators instance of Node Operators Registry
-    * @param _treasury treasury contract
-    * @param _insuranceFund insurance fund contract
-    * NB: by default, staking and the whole Lido pool are in paused state
-    */
-    function initialize(
-        IDepositContract _depositContract,
-        address _oracle,
-        INodeOperatorsRegistry _operators,
-        address _treasury,
-        address _insuranceFund
-    )
-        public onlyInit
-    {
-        NODE_OPERATORS_REGISTRY_POSITION.setStorageAddress(address(_operators));
-        DEPOSIT_CONTRACT_POSITION.setStorageAddress(address(_depositContract));
+    /// @dev version of contract
+    bytes32 internal constant CONTRACT_VERSION_POSITION = keccak256("lido.Lido.contractVersion");
 
-        _setProtocolContracts(_oracle, _treasury, _insuranceFund);
+    /**
+     * @dev As AragonApp, Lido contract must be initialized with following variables:
+     * @param _oracle oracle contract
+     * @param _treasury treasury contract
+     * @param _stakingRouter Staking router contract
+     * @param _dsm Deposit security module contract
+     * NB: by default, staking and the whole Lido pool are in paused state
+     */
+    function initialize(address _oracle, address _treasury, address _stakingRouter, address _dsm) public onlyInit {
+        _setProtocolContracts(_oracle, _treasury);
 
+        _initialize_v2(_stakingRouter, _dsm);
```

Comment:
Add comments that if we are deploying the protocol from scratch there are circular dependencies introduced (dsm and staking router): on init stage set dsm and staking router as 0x0, and afterwards use setters for set them correctly